### DISCUSSION TITLE
part-ii/2017-26/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#374] [Question comments]
Part II 2017 Automata and Formal Languages: Paper 1, Section II, $11 H$ 

https://questions.tripos.org/part-ii/2017-26/

**Comment by artie2000** [comment]
I found this a bit hard to justify fully, but the idea is to write down algorithms decoding the DFA and then performing certain operations on it.

<details>
<summary>The key idea:</summary>

A DFA with $k$ states has an infinite language if and only if it accepts a word $w$ of length $k\leq |w|<2k$.

Then checking for finiteness is easy; to compute the (finite) size, check all words of length at most $k-1$.

</details>
